### ISSUE TITLE
decorators that are instances of classes have methods stripped when access through context

### ISSUE THREAD

**Issue by ethanniser** [elysiajs/elysia#143]
```ts
import { Elysia } from "elysia";

class Test {
  readonly name = "test";
  public foo() {
    return this.name;
  }
}

const test = new Test();

console.log(test);

export const ctx = new Elysia().decorate("test", test);

const app = new Elysia()
  .use(ctx)
  .get("/", ({ test }) => {
    console.log(test);
    console.log(test.foo());
  })
  .listen(3002);

console.log(`app is listening on ${app.server?.hostname}:${app.server?.port}`);

app.handle(new Request("http://localhost:3002/"));
```
```
$ bun run --hot src/main.ts
[0.02ms] ".env"
{
  name: "test",
  foo: [Function: foo]
}
app is listening on localhost:3002
{
  name: "test"
}
TypeError: test.foo is not a function. (In 'test.foo()', 'test.foo' is undefined)
```

very very strange behavior, and only happens with classes

if you replace
```ts
const test = new Test();
```
with
```ts
const test = {
  name: "test",
  foo() {
    return this.name;
  },
};
```

everything works fine
```
$ bun run --hot src/main.ts
[0.03ms] ".env"
{
  name: "test",
  foo: [Function: foo]
}
app is listening on localhost:3002
{
  name: "test",
  foo: [Function: foo]
}
test
```

**Comment by HKGx** [comment]
This issue stems from Mergician, if you call `.use`, it tries to merge your decorators but somehow fails. This probably can be alleviated, but I'd like to create an issue at  Mergician

**Comment by HKGx** [comment]
https://github.com/jhildenbiddle/mergician/issues/4

**Comment by SaltyAom** [comment]
Should be fixed in #103.

Let me know if the problem stills persists
